micro_dl/config/_torch_config.py

```diff
@@ -0,0 +1,99 @@
+from typing import Set, Union, Literal
+from pydantic import BaseModel, validator
+from typing import Optional
+import os
+from torch.cuda import device_count
+
+class _TorchModel(BaseModel):
+    """
+    Configuration validation skeleton for model subconfig
+    of torch master config.
+    """
+    architecture: Literal['2.5D', '2D']
+    in_channels: int
+    out_channels: int
+    residual: bool
+    task: Literal['reg','seg']
+    model_dir: str
+    
+    @validator('model_dir')
+    def valid_path(cls, path):
```

Comment:
I'm actually not sure how to avoid the boiler plate repetition in pydantic. I'm very much a beginner user myself. I had originally written only single functions but I couldn't get pydantic to pull the input values correctly, so this is the hacky fix I settled for. I think more time needs to be spent on this code; let's push this back to the pydantic branch PR when it's ready to merge.